### ISSUE TITLE
CI: remove "experimental" checks, as they may be redundant

### ISSUE THREAD

**Issue by thaJeztah** [docker/cli#5140]
### Description

While working on https://github.com/docker/cli/pull/5139, I noticed that CI was green even if we didn't enable experimental on the daemon.

I was curious how the `experimental` and `non-experimental` e2e tests differed, and after a cursory glance, it may be only for a _single_ test; https://github.com/docker/cli/blob/591bd174241555b26fb5e92de5d7a9da094523c1/e2e/image/build_test.go#L128

If that's correct, then it would mean we run _all_ e2e tests twice, but only for a single test; we can cut the number of checks in half if we don't do so;

- Always enable `experimental` on the daemon (we already do so!)
- Run the tests and let the "experimental" one be skipped
- Rejoice!

**Comment by Benehiko** [comment]
>  - Always enable experimental on the daemon (we already do so!)
>  - Run the tests and let the "experimental" one be skipped
>  - Rejoice!

@thaJeztah So if we are always running an experimental daemon, do you mean to say we should disable the non-experimental run?

**Comment by thaJeztah** [comment]
Yeah, if it's indeed only that one tests I saw being different, we could run all e2e tests with "experimental" enabled (unconditionally), and just have a single e2e suite (that runs both "experimental" and "non-experimental" tests)
